server/service/transaction_service.rs

```diff
@@ -425,20 +425,29 @@ impl TransactionService {
 
         let transaction = match transaction_type {
             typedb_protocol::transaction::Type::Read => {
-                Transaction::Read(TransactionRead::open(database, transaction_options).map_err(|typedb_source| {
-                    TransactionServiceError::TransactionFailed { typedb_source }.into_error_message().into_status()
-                })?)
+                let transaction = spawn_blocking(move ||
```

Comment:
We sometimes have to wait for the watermark, so better to run this in spawn_blocking